File: frontend/lib/models/milestone.model.dart
class Milestone {
  final String id;
  final String milestoneType;
  final DateTime dateAchieved;
  final String? notes;

  Milestone({
    required this.id,
    required this.milestoneType,
    required this.dateAchieved,
    this.notes,
  });

  factory Milestone.fromJson(Map<String, dynamic> json) {
    return Milestone(
      id: json['_id'],
      milestoneType: json['milestoneType'],
      dateAchieved: DateTime.parse(json['dateAchieved']),
      notes: json['notes'],
    );
  }
}

File: backend/models/milestone.model.js
const mongoose = require("mongoose");

const MilestoneSchema = new mongoose.Schema({
  parentId: { type: mongoose.Schema.Types.ObjectId, ref: "Parent", required: true },
  milestoneType: { type: String, required: true },
  dateAchieved: { type: Date, required: true },
  notes: { type: String, default: "" }
}, { timestamps: true });

module.exports = mongoose.model("Milestone", MilestoneSchema);
